adds Hashi approle examples for secrets helper (#3584)
chef/chef-web-docs: content/infra_language/secrets.md

      service: :hashi_vault,
      config: {
        vault_addr: 'vault.example.com',
        auth_method: :token,
        token: '123456'
      })
```

#### Fetching Secrets From HashiCorp Vault Using AppRole Authentication

Fetching secret data using an AppRole ID and an associated AppRole Secret ID:

```ruby
secret(name: 'secret/example',
      service: :hashi_vault,
      config: {
        vault_addr: 'vault.example.com',
        auth_method: :approle,
        approle_id: "11111111-abcd-1111-abcd-111111111111",
        approle_secret_id: "22222222-abcd-2222-abcd-222222222222"
      })
```

> Fetching secret data using a token along with an AppRole name which will create a Secret ID associated to the AppRole:

Reviewer: ```suggestion
Fetching secret data using a token and an AppRole name creates a Secret ID associated with that AppRole:
```

Author: @kagarmoe cool!  I just updated this and rebased as well since it was out of date
